File: articles/leadership-styles-organizational-performance/go/leadership_style_score.go
// Leadership Styles and Organizational Performance
// Simple Go scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment, productivity ranking,
// leadership-loyalty scoring, dissent tracking, or individual decisions.

package main

import "fmt"

type LeadershipStyleProfile struct {
	ParticipativeBehavior float64
	StructuralGuidance    float64
	CommunicationQuality  float64
	TrustGeneration       float64
	DevelopmentalSupport  float64
	PsychologicalSafety   float64
	DecisionFit           float64
	CulturalAlignment     float64
	ControlPressure       float64
	AmbiguityRisk         float64
	DisengagementPressure float64
}

func LeadershipEffectivenessScore(p LeadershipStyleProfile) float64 {
	return 0.11*p.ParticipativeBehavior +
		0.12*p.StructuralGuidance +
		0.13*p.CommunicationQuality +
		0.13*p.TrustGeneration +
		0.11*p.DevelopmentalSupport +
		0.12*p.PsychologicalSafety +
		0.11*p.DecisionFit +
		0.10*p.CulturalAlignment -
		0.08*p.ControlPressure -
		0.07*p.AmbiguityRisk -
		0.07*p.DisengagementPressure
}

func main() {
	profile := LeadershipStyleProfile{82, 80, 83, 81, 79, 80, 82, 81, 18, 22, 20}
	fmt.Printf("Synthetic leadership effectiveness score: %.2f\n", LeadershipEffectivenessScore(profile))
	fmt.Println("Responsible-use reminder: institutional learning only.")
}

File: articles/leadership-styles-organizational-performance/cpp/leadership_style_score.cpp
/*
Leadership Styles and Organizational Performance
Simple C++ scoring demonstration.

Responsible-use scope:
Synthetic-data research, methods demonstration, institutional learning,
and reproducible workflows only. Not for employment, productivity ranking,
leadership-loyalty scoring, dissent tracking, or individual worker decisions.
*/

#include <iostream>

struct LeadershipStyleProfile {
    double participative_behavior;
    double structural_guidance;
    double communication_quality;
    double trust_generation;
    double developmental_support;
    double psychological_safety;
    double decision_fit;
    double cultural_alignment;
    double control_pressure;
    double ambiguity_risk;
    double disengagement_pressure;
};

double leadership_effectiveness_score(const LeadershipStyleProfile& p) {
    return
        0.11 * p.participative_behavior +
        0.12 * p.structural_guidance +
        0.13 * p.communication_quality +
        0.13 * p.trust_generation +
        0.11 * p.developmental_support +
        0.12 * p.psychological_safety +
        0.11 * p.decision_fit +
        0.10 * p.cultural_alignment -
        0.08 * p.control_pressure -
        0.07 * p.ambiguity_risk -
        0.07 * p.disengagement_pressure;
}

int main() {
    LeadershipStyleProfile profile{82, 80, 83, 81, 79, 80, 82, 81, 18, 22, 20};
    std::cout << "Synthetic leadership effectiveness score: "
              << leadership_effectiveness_score(profile) << "\n";
    std::cout << "Responsible-use reminder: institutional learning only.\n";
    return 0;
}
